Output summary

DOH-FAA3303 opened the pull request:
## 📑 Description
<!-- Add a brief description of the pr -->

<!-- You can also choose to add a list of changes and if they have been completed or not by using the markdown to-do list syntax
- [ ] Not Completed
- [x] Completed
-->

closes #52 

this will add an output summary of the fuzzy matching functions to show the user how much data went through the script and how many records matched/didn't match. it also provides error handling: if any data leaks are found between the original submissions dataframe and the combined outputs it will error the script.

example of a success
```python
Success: No data leaks detected. Insert victory cigar
2 exact matches
4 fuzzy matched
4 no match found
0 records without demo

Total unique persons processed:  10 
Original submissions to fuzzy (including no_match rematch attempt): 10
```

example of a failure
```python
ERROR!! Fuzzy data leaked. Data are in submissions_to_fuzzy_df that cannot be found in fuzzy outputs
Total processed:  8 
Original submissions to fuzzy (including no_match rematch attempt): 10
```

## How to Test

If data fails, the VS Code or Positron debugger is key. Set a breakpoint in the dataframe_matcher.py script and you can open up internal scopes of the functions and examine the temporary dataframes to determine where the data leak is occurring.

<!--
Please describe the steps to test your changes. Include any relevant testing instructions or guidelines.
-->

## ✅ Checks
<!-- Make sure your pr passes the CI checks and do check the following fields as needed - -->
-   [x] I have **reviewed** the pull request for any sensitive data, including text and images.
-   [x] I have read and followed the contributing guidelines.
-   [x] I have checked my code for any issues.
-   [ ] I have updated the documentation (if applicable).
-   [ ] I have added tests to cover my changes (if applicable).
-   [ ] All tests are passing (if applicable).
-   [x] My changes do not introduce any new warnings.
-   [x] I have run the code locally and verified the changes work as expected.

## Screenshots (if applicable)

<!--
If your pull request includes images or screenshots, **please confirm** that no sensitive information is present. If unsure, remove or blur any potentially sensitive content.
-->

DOH-PXC5303 commented on src/wadoh_raccoon/dataframe_matcher.py:
```diff
@@ -380,6 +380,71 @@ def fuzzy_match(self, dob_match) -> (pl.DataFrame, pl.DataFrame):
+        # anti_join outputs to see if any data is missing from the outputs from the original df
+        check_data_leaks = (
+            submissions_to_fuzzy_df
+            .join(all_outputs, on="submission_number",how="anti")
```
Instead of hard-coded "submission_number", use self.key

DOH-PXC5303 commented on src/wadoh_raccoon/dataframe_matcher.py:
```diff
@@ -380,6 +380,71 @@ def fuzzy_match(self, dob_match) -> (pl.DataFrame, pl.DataFrame):
+        # try the opposite anti join
+        check_data_leaks_reverse = (
+            all_outputs
+            .join(submissions_to_fuzzy_df, on="submission_number",how="anti")
```
Instead of "submission_number", use self.key

DOH-PXC5303 reviewed (changes requested):
Looks good. I added comments about using self.key instead of submission_number. This failed for me testing data without that field (where instead the key was created as "__key__")

DOH-FAA3303 commented:
> Looks good. I added comments about using self.key instead of submission_number. This failed for me testing data without that field (where instead the key was created as "__key__")

oops. good call, i replaced them in a new commit
